pkg/operator/admission/webhook.go

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package admission
+
+import (
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/operator/ceph/csi"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"os"
+)
+
+const (
+	appName            = "rook-admission-controller"
+	secretVolumeName   = "webhook-certs"
+	secretMountPath    = "/etc/webhook"
+	serviceAccountName = "rook-admission-controller-sa"
+	storageFlag        = "ceph"
```

Comment:
More appropriate name is `storageBackend`.
```suggestion
	storageBackend        = "ceph"
```

In reality, this code is designed to support other backends in the future, not just Ceph, but Ceph is very hardcoded at the moment. This isn't necessarily bad, but it may actually be better to suggest this more with the naming.

```suggestion
	// StorageBackendCeph is the value used to select the admission controller for Rook's Ceph backend.
	StorageBackendCeph        = "ceph"
```